subsystems/viamserver/viamserver.go

```diff
@@ -150,16 +149,19 @@ func (s *viamServer) Start(ctx context.Context) error {
 		defer s.mu.Unlock()
 		s.running = false
 		s.logger.Infof("%s exited", SubsysName)
-		if err != nil {
-			s.logger.Error(errw.Wrap(err, "error while getting process status"))
-		}
-		if s.cmd.ProcessState != nil {
-			s.lastExit = s.cmd.ProcessState.ExitCode()
-			if s.lastExit != 0 {
-				s.logger.Errorf("non-zero exit code: %d", s.lastExit)
-			}
-		}
+		// Only log errors from Wait() or the exit code of the process state if subsystem
+		// exited unexpectly (was not stopped by agent and is therfore still marked as
+		// shouldRun).
```

Comment:
We were getting unnecessary error logs here when restarting. A non-zero exit code when we sent SIGQUIT to `viam-server` is expected.